src/views/Pools/components/LockedPool/Common/BalanceField.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState, useCallback, memo, Dispatch, SetStateAction } from 'react'
+
+import { Text, Flex, Image, BalanceInput, Slider, Button } from '@pancakeswap/uikit'
+import styled from 'styled-components'
+import { BIG_TEN } from 'utils/bigNumber'
+
+import { getFullDisplayBalance } from 'utils/formatBalance'
+import BigNumber from 'bignumber.js'
+import { useTranslation } from 'contexts/Localization'
+
+const StyledButton = styled(Button)`
+  flex-grow: 1;
+`
+
+interface PropsType {
+  stakingAddress: string
+  stakingSymbol: string
+  stakingDecimals: number
+  lockedAmount: number
+  stakingMax: BigNumber
+  setLockedAmount: Dispatch<SetStateAction<number>>
+  usedValueStaked: number | undefined
+}
+
+const BalanceField: React.FC<PropsType> = ({
+  stakingAddress,
+  stakingSymbol,
+  stakingDecimals,
+  lockedAmount,
+  stakingMax,
+  setLockedAmount,
+  usedValueStaked,
+}) => {
+  const { t } = useTranslation()
+  const [percent, setPercent] = useState(0)
+
+  const handleStakeInputChange = useCallback(
+    (input: string) => {
+      if (input) {
+        const convertedInput = new BigNumber(input).multipliedBy(BIG_TEN.pow(stakingDecimals))
+        const percentage = Math.floor(convertedInput.dividedBy(stakingMax).multipliedBy(100).toNumber())
+        setPercent(percentage > 100 ? 100 : percentage)
+      } else {
+        setPercent(0)
+      }
+      setLockedAmount(Number(input))
```

Comment:
I think this should be  `setLockedAmount(input)`
If direct change to Number, can't typing decimal in input.

Reply:
Thasnk. Good catch.